File: frontend/src/services/complaintService.js
import apiClient from './apiClient';

export const complaintService = {
  getAllComplaints: async (params = {}) => {
    const response = await apiClient.get('/api/complaints', { params });
    return response.data;
  },
  getComplaintById: async (id) => {
    const response = await apiClient.get(`/api/complaints/${id}`);
    return response.data;
  },
  getStudentComplaints: async (studentId) => {
    const response = await apiClient.get(`/api/complaints/student/${studentId}`);
    return response.data;
  },
  createComplaint: async (data) => {
    const response = await apiClient.post('/api/complaints', data);
    return response.data;
  },
  updateStatus: async (id, payload) => {
    const response = await apiClient.patch(`/api/complaints/${id}/status`, payload);
    return response.data;
  },
  deleteComplaint: async (id) => {
    const response = await apiClient.delete(`/api/complaints/${id}`);
    return response.data;
  },
};

File: backend/src/main/java/com/yourorg/appname/service/ComplaintService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.ComplaintRequest;
import com.yourorg.appname.dto.response.ComplaintResponse;
import java.util.List;

public interface ComplaintService {
    List<ComplaintResponse> getAllComplaints(String status, String category, String priority);
    ComplaintResponse getComplaintById(Long id);
    ComplaintResponse getComplaintByTicket(String ticketNumber);
    List<ComplaintResponse> getComplaintsByStudentId(Long studentId);
    ComplaintResponse createComplaint(ComplaintRequest request, String username);
    ComplaintResponse updateComplaintStatus(Long id, String status, String resolutionNotes, String assignedStaff);
    void deleteComplaint(Long id);
}
